master/internal/api_user.go

```diff
@@ -441,7 +442,16 @@ func (a *apiServer) PatchUser(
 		}
 
 		updatedUser.Remote = *req.User.Remote
+		willBeRemote = updatedUser.Remote
 		insertColumns = append(insertColumns, "remote")
+
+		// Clear password when we convert a remote user into non remote without password specified.
+		if !updatedUser.Remote && req.User.Password == nil {
+			if err := updatedUser.UpdatePasswordHash(""); err != nil {
+				return nil, errors.Wrap(err, "error hashing password")
+			}
+			insertColumns = append(insertColumns, "password_hash")
+		}
```

Comment:
set the password to "NoPasswordLogin" here when we make a user "remote". the "remote users can't login through POST /login" code is OK to keep but they should also have "NoPasswordLogin".

that's already how they get setup when they are created remote, so patching them that way shouldn't result in a different state.